### PR TITLE
chore(web): connects gesture-recognizer tests for Web CI testing 🐵

### PR DESCRIPTION
**Pull request by jahorton**
Up until now, the gesture-recognizer's automated tests haven't actually been run by our CI.  This PR seeks to fix that... along with a few issues that were causing certain test configurations to fail.

While this was originally included as part of #9267, it appears best to have it implemented as its own PR due to "fun" complications, mostly surrounding testing on real Android devices via BrowserStack.  In particular...

https://github.com/keymanapp/keyman/blob/30c23760ae010bfcd59210ed9a533b9ef8746ae9/common/web/gesture-recognizer/src/test/auto/browser/cases/recordedCoordSequences.js#L48-L55

I spent pretty much all of yesterday afternoon trying every angle I could think of to work around the issue described there that _didn't_ involve coercing the "mouse-based" sequences to use "touch-based" simulation.  Absolutely nothing worked.  Some details from the investigation:
- While the main 'test fixture' itself _is_ pixel-aligned, the "host-fixture" element within it was _not_ aligned on the BrowserStack test device.  I'm not sure why this was the case, as the issue does not repro locally under Chrome remote-device emulation.
- Any attempt to offset the 'test fixture' to pixel-align the "host-fixture" element failed - `getClientBoundingRect()`'s returned `DOMRect` reported absolutely no change in position when adding subpixel offsets to try correcting the issue.

Rather than throw even more time at the problem, the simple solution that allows us to move on for now is to simply force the sequences to use touch-events when on the affected test config.

@keymanapp-test-bot skip

### CODE REVIEW

File: common/web/gesture-recognizer/src/test/resources/json/receiver/embeddedBorderCancel.json [hunk 1]
```diff
@@ -143,12 +143,12 @@
               },
               {
                 "targetX": 118,
-                "targetY": 4,
+                "targetY": 5,
```
Comment (jahorton) [inline common/web/gesture-recognizer/src/test/resources/json/receiver/embeddedBorderCancel.json:146]: The 'safe bound' applied during the check is ` < 4`.  Unfortunately, since there tend to be subpixel calculations when testing on real Android test devices, it's possible to get something akin to `3.9999` (rather than a properly-integer `4`) due to floating-point approximations, and that affects the check.

This correction provides an ample "fudge-factor" to buffer against such numerical approximation errors so that they don't adversely fail the corresponding automated test.

File: common/web/gesture-recognizer/src/test/auto/browser/cases/recordedCoordSequences.js [hunk 2]
```diff
@@ -39,6 +44,23 @@ describe("Layer one - DOM -> InputSequence", function() {
       let resultPromise;
 
       let playbackEngine = new InputSequenceSimulator(this.controller);
+
+      // **********************************
+      // Android-Chrome sequence simulation does not allow fractional values in MouseEvent clientX/clientY...
+      // and it's impossible to coerce elements to have proper pixel alignment, it seems.  Fractional
+      // values are fine for touch events `Touch` objects, though.
```
Comment (jahorton) [inline common/web/gesture-recognizer/src/test/auto/browser/cases/recordedCoordSequences.js:51]:
```suggestion
      // values are fine for touch events' `Touch` objects, though.
```

File: common/web/gesture-recognizer/src/test/auto/browser/cases/recordedCoordSequences.js [hunk 3]
```diff
@@ -39,6 +44,23 @@ describe("Layer one - DOM -> InputSequence", function() {
       let resultPromise;
 
       let playbackEngine = new InputSequenceSimulator(this.controller);
+
+      // **********************************
+      // Android-Chrome sequence simulation does not allow fractional values in MouseEvent clientX/clientY...
```
Comment (mcdurdin) [inline common/web/gesture-recognizer/src/test/auto/browser/cases/recordedCoordSequences.js:49]: I think real mouse events are always pixel aligned on all platforms, aren't they? So I'm not really understanding why this is so big an issue?

**Review by mcdurdin — APPROVED** [review]

File: common/web/gesture-recognizer/src/test/auto/browser/cases/recordedCoordSequences.js (hunk 3 above)
Comment (jahorton) [reply common/web/gesture-recognizer/src/test/auto/browser/cases/recordedCoordSequences.js:49]: The affected automated tests check, among other things, how boundary conditions are handled and respected over the lifetime of the touchpath.  If the reproduced path's alignment is off, it's possible for a sequence that was in-bounds originally to be misaligned enough to go partly out of bounds... or even just out of bounds early.
